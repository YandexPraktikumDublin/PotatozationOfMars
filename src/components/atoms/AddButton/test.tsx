import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { AddButton } from '.'

describe('<AddButton />', () => {
  it('should renders correct <AddButton />', () => {
    const onClick = jest.fn()

    const wrapper = shallow(<AddButton onClick={onClick} />)

    wrapper.simulate('click')

    expect(onClick).toHaveBeenCalledTimes(1)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has className from className prop', () => {
    const onClick = jest.fn()
    const className = 'test-class-name'

    const wrapper = shallow(
      <AddButton onClick={onClick} className={className} />
    )

    expect(wrapper.prop('className')).toMatch(className)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
