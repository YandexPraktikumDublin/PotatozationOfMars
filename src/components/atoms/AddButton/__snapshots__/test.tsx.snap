// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddButton /> should has className from className prop 1`] = `
<button
  className="test-class-name"
  onClick={[MockFunction]}
>
  <svg
    height="20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="fill-primary dark:fill-white"
      d="M10 .43C4.723.43.43 4.723.43 10s4.293 9.57 9.57 9.57 9.57-4.293 9.57-9.57S15.277.43 10 .43zm0 17.547c-4.398 0-7.977-3.579-7.977-7.977S5.602 2.023 10 2.023 17.977 5.602 17.977 10 14.398 17.977 10 17.977z"
    />
    <path
      className="fill-primary dark:fill-white"
      d="M13.969 9.203h-3.172V6.031A.795.795 0 0010 5.234a.795.795 0 00-.797.797v3.172H6.031a.795.795 0 00-.797.797c0 .441.356.797.797.797h3.172v3.172c0 .441.356.797.797.797a.795.795 0 00.797-.797v-3.172h3.172a.795.795 0 00.797-.797.795.795 0 00-.797-.797z"
    />
  </svg>
</button>
`;

exports[`<AddButton /> should renders correct <AddButton /> 1`] = `
<button
  className=""
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <svg
    height="20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="fill-primary dark:fill-white"
      d="M10 .43C4.723.43.43 4.723.43 10s4.293 9.57 9.57 9.57 9.57-4.293 9.57-9.57S15.277.43 10 .43zm0 17.547c-4.398 0-7.977-3.579-7.977-7.977S5.602 2.023 10 2.023 17.977 5.602 17.977 10 14.398 17.977 10 17.977z"
    />
    <path
      className="fill-primary dark:fill-white"
      d="M13.969 9.203h-3.172V6.031A.795.795 0 0010 5.234a.795.795 0 00-.797.797v3.172H6.031a.795.795 0 00-.797.797c0 .441.356.797.797.797h3.172v3.172c0 .441.356.797.797.797a.795.795 0 00.797-.797v-3.172h3.172a.795.795 0 00.797-.797.795.795 0 00-.797-.797z"
    />
  </svg>
</button>
`;
